tfhe/src/high_level_api/booleans/tests.rs

```diff
@@ -3,21 +3,346 @@
+    assert_eq!(r.decrypt(key), false);
+    assert_degree_is_ok(&r);
+
+    let r = ttrue.eq(ffalse);
+    assert_eq!(r.decrypt(key), false);
+    assert_degree_is_ok(&r);
+
+    let r = ttrue.eq(ttrue);
+    assert_eq!(r.decrypt(key), true);
+    assert_degree_is_ok(&r);
+
+    let r = ffalse.eq(ffalse);
+    assert_eq!(r.decrypt(key), true);
+    assert_degree_is_ok(&r);
+}
+
+fn scalar_eq_truth_table(ttrue: &FheBool, ffalse: &FheBool, key: &ClientKey) {
+    assert_degree_is_ok(ttrue);
+    assert_degree_is_ok(ffalse);
+
+    let r = ffalse.eq(true);
+    assert_eq!(r.decrypt(key), false);
+    assert_degree_is_ok(&r);
+
+    let r = ttrue.eq(false);
+    assert_eq!(r.decrypt(key), false);
+    assert_degree_is_ok(&r);
 
-    let (my_keys, server_keys) = generate_keys(config);
+    let r = ttrue.eq(true);
+    assert_eq!(r.decrypt(key), true);
+    assert_degree_is_ok(&r);
 
-    set_server_key(server_keys);
-    my_keys
+    let r = ffalse.eq(false);
+    assert_eq!(r.decrypt(key), true);
+    assert_degree_is_ok(&r);
+}
+
+fn ne_truth_table(ttrue: &FheBool, ffalse: &FheBool, key: &ClientKey) {
+    assert_degree_is_ok(ttrue);
+    assert_degree_is_ok(ffalse);
+
+    let r = ffalse.ne(ttrue);
+    assert_eq!(r.decrypt(key), true);
+    assert_degree_is_ok(&r);
+
+    let r = ttrue.ne(ffalse);
+    assert_eq!(r.decrypt(key), true);
+    assert_degree_is_ok(&r);
+
+    let r = ttrue.ne(ttrue);
+    assert_eq!(r.decrypt(key), false);
+    assert_degree_is_ok(&r);
+
+    let r = ffalse.ne(ffalse);
+    assert_eq!(r.decrypt(key), false);
+    assert_degree_is_ok(&r);
+}
+
+fn scalar_ne_truth_table(ttrue: &FheBool, ffalse: &FheBool, key: &ClientKey) {
+    assert_degree_is_ok(ttrue);
+    assert_degree_is_ok(ffalse);
+
+    let r = ffalse.ne(true);
+    assert_eq!(r.decrypt(key), true);
+    assert_degree_is_ok(&r);
+
+    let r = ttrue.ne(false);
+    assert_eq!(r.decrypt(key), true);
+    assert_degree_is_ok(&r);
+
+    let r = ttrue.ne(true);
+    assert_eq!(r.decrypt(key), false);
+    assert_degree_is_ok(&r);
+
+    let r = ffalse.ne(false);
+    assert_eq!(r.decrypt(key), false);
+    assert_degree_is_ok(&r);
+}
+
+fn compressed_bool_test_case(setup_fn: impl FnOnce() -> (ClientKey, Device)) {
+    let (cks, sks_device) = setup_fn();
+
+    let cttrue = CompressedFheBool::encrypt(true, &cks);
+    let cffalse = CompressedFheBool::encrypt(false, &cks);
+
+    let a = FheBool::from(cttrue);
+    let b = FheBool::from(cffalse);
+
+    assert_eq!(a.current_device(), sks_device);
+    assert_eq!(b.current_device(), sks_device);
+
+    assert_eq!(a.decrypt(&cks), true);
+    assert_eq!(b.decrypt(&cks), false);
+}
+
+fn compact_bool_test_case(setup_fn: impl FnOnce() -> (ClientKey, Device)) {
+    let (cks, sks_device) = setup_fn();
+    let cpk = CompactPublicKey::new(&cks);
+
+    let cttrue = CompactFheBool::encrypt(true, &cpk);
+    let cffalse = CompactFheBool::encrypt(false, &cpk);
+
+    let a = cttrue.expand();
+    let b = cffalse.expand();
```

Comment:
can we check the degree here as well in the various compressed/compact etc. encryptions ? I believe you have the helper function to do that

I'm guessing PK as well ?

This is getting annoying, sorry :/

Reply:
done